modules/modal/react/lib/Modal.tsx

```diff
@@ -70,22 +147,80 @@ export default class Modal extends React.Component<ModalProps> {
   };
 
   public render() {
-    const {open, handleClose, padding, width, heading, transformOrigin, ...elemProps} = this.props;
+    const {
+      open,
+      handleClose,
+      padding,
+      width,
+      heading,
+      testId,
+      children,
+      closeOnEscape,
+      firstFocusRef: firstFocusableRef,
+      ...elemProps
+    } = this.props;
     return (
       open && (
-        <Container onClick={e => this.handleOutsideClick(handleClose, e)} {...elemProps}>
-          <Popup
-            popupRef={this.modalRef}
-            width={width}
-            heading={heading}
-            handleClose={handleClose}
-            padding={padding}
-            transformOrigin={transformOrigin}
-          >
-            {this.props.children}
-          </Popup>
-        </Container>
+        <FocusTrap
+          focusTrapOptions={{
+            initialFocus: () =>
+              onInitialFocus(this.modalRef.current, firstFocusableRef && firstFocusableRef.current),
+          }}
+        >
+          <Container onClick={e => this.handleOutsideClick(handleClose, e)} {...elemProps}>
+            <Popup
+              popupRef={this.modalRef}
+              width={width}
+              heading={heading}
+              handleClose={handleClose}
+              padding={padding}
+              transformOrigin={transformOrigin}
+              data-testid={testId}
+            >
+              {children}
+            </Popup>
+          </Container>
+        </FocusTrap>
       )
     );
   }
 }
+
+/**
+ * Convenience hook to set up props for both a target and the modal component.
+ * @returns An object containing convenience variables to mix into component parts of a Modal
+ * @example
+ * const myComponent = () => {
+ *   const {targetProps, modalProps, closeModal} = useModal();
+ *
+ *   return (
+ *     <>
+ *       <Button {...targetProps}>Delete Item</Button>
+ *       <Modal heading='Delete Item' {...modalProps}>
+ *         Are you sure?
+ *         <Button onClick={closeModal}>Cancel</Button>
+ *       </Modal>
+ *     </>
+ *   )
+ * }
+ */
+export function useModal() {
+  const [open, setOpen] = React.useState(false);
+
+  return {
+    targetProps: {
+      onClick(event: React.SyntheticEvent<HTMLElement>) {
```

Comment:
I guess I don't need to be so explicit on the signature here... I used to save the element to refocus on close. `handleClose` doesn't have any restrictions. Thoughts?